onnxruntime/python/tools/quantization/save_activations.py

```diff
@@ -0,0 +1,156 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft, Intel Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""Utilities to run a given ONNX model, while saving input/output tensors of
+eligible operator nodes.
+
+A use case is to debug quantization induced accuracy drop. An AI engineer can
+run the original float32 model and the quantized model with the same inputs,
+then compare the corresponding activations between the two models to find
+where the divergence is. 
+
+Example Usage:
+
+.. code-block:: python
+    class ExampleDataReader(CalibrationDataReader):
+        def __init__(self):
+            ...
+        def get_next(self):
+            ...
+
+    input_data_reader = ExampleDataReader()
+
+    aug_model = aug_model_for_act_saving(path_to_onnx_model)
+    augmented_model_path = str(Path(self._tmp_model_dir.name).joinpath("augmented_model.onnx"))
+    onnx.save(
+        aug_model,
+        augmented_model_path,
+        save_as_external_data=False,
+    )
+
+    tensor_dict = run_collect_activations(augmented_model_path, data_reader)
+    # tensor_dict points to a dictionary where the keys are tensor names and each value
+    # is a list of tensors, one from each model run
+
+"""
+
+import uuid
+
+from typing import Union,List
+from pathlib import Path
+
+import numpy
+import onnx
+from onnx import ModelProto, TensorProto, helper, numpy_helper
+
+import onnxruntime
+from .calibrate import CalibrationDataReader, CalibraterBase
+from .quant_utils import clone_model_with_shape_infer
+
+
+def aug_model_for_act_saving(
```

Comment:
readability: Avoid abbreviations - they add cognitive load to readers.

> Do not use abbreviations that are ambiguous or unfamiliar to readers outside the project, and do not abbreviate by deleting letters within a word.

https://google.github.io/styleguide/pyguide.html#316-naming
https://google.github.io/styleguide/cppguide.html#General_Naming_Rules

Reply:
fixed